library/sinks/MongoDB.ts

```diff
@@ -137,17 +139,28 @@ export class MongoDB implements Wrapper {
   private inspectOperation(
     operation: string,
     args: unknown[],
-    collection: Collection
+    collection: Collection,
+    filterPosition = 0
   ): InterceptorResult {
     const context = getContext();
 
     if (!context) {
       return undefined;
     }
 
-    if (args.length > 0 && isPlainObject(args[0])) {
-      const filter = args[0];
+    let filter: unknown;
+
+    if (filterPosition === 0 && args.length > 0 && isPlainObject(args[0])) {
+      filter = args[0];
+    } else if (
+      filterPosition === 1 &&
+      args.length > 1 &&
+      isPlainObject(args[1])
+    ) {
+      filter = args[1];
+    }
```

Comment:
Maybe we can write this a bit simpler?

e.g. loop over all the args? And if it's a plainObject, inspect the filter?

Reply:
Passed objects do not necessarily have to be a filter, they can also be e.g. options.